`clever deploy` on already pushed commit

## Context

Hi,

If a commit B had been pushed and deployed. Then a A commit which is older than B is redeployed from Console. And finally `$ clever deploy` to redeploy A. There is an error message specifying that the current clever application is up-to-date:                                                                                                                                                                     

`[ERROR] The clever-cloud application is up-to-date. Try 'clever restart' to restart the application`

We should edit the message or create an interaction to specify the last most recent commit by default.

## TODO

* [ ] When `clever deploy` fails and advise to do a `clever restart` it should mention the commit id if it's necessary.

--- a/src/commands/deploy.js
+++ b/src/commands/deploy.js
@@ -29,19 +29,30 @@ function deploy (api, params) {
     .toProperty();
 
   const s_commitIdToPush = Git.getCommitId(branchName).toProperty();
-  const s_remoteCommitId = s_remote
+  const s_remoteHeadCommitId = s_remote
     .map((remote) => Git.getRemoteName(remote))
     .flatMapLatest((remoteName) => Git.getRemoteCommitId(remoteName))
     .toProperty();
+  const s_deployedCommitId = s_appData
+    .flatMapLatest(({ app_id, app_orga }) => Application.get(api, app_id, app_orga))
+    .flatMapLatest(({ commitId }) => commitId)
+    .toProperty();
 
-  const s_allLogs = Bacon.combineAsArray(s_commitIdToPush, s_remoteCommitId, s_remote, s_branchRefspec)
-    .flatMapLatest(([commitIdToPush, remoteCommitId, remote, branchRefspec]) => {
+  const s_allLogs = Bacon.combineAsArray(s_commitIdToPush, s_remoteHeadCommitId, s_deployedCommitId)
+    .flatMapLatest(([commitIdToPush, remoteCommitId, deployedCommitId]) => {
       if (commitIdToPush === remoteCommitId) {
-        return new Bacon.Error('The clever-cloud application is up-to-date. Try `clever restart` to restart the application');
+        const upToDateMessage = `The clever-cloud application is up-to-date. Try this command to restart the application:`;
+        if (commitIdToPush !== deployedCommitId) {
+          return new Bacon.Error(`${upToDateMessage}\nclever restart --commit ${commitIdToPush}`);
+        }
+        return new Bacon.Error(`${upToDateMessage}\nclever restart`);
       }
-      Logger.println("Pushing source code to Clever Cloud.");
+      return Bacon.combineAsArray(s_remote, s_branchRefspec);
+    })
+    .flatMapLatest(([remote, branchRefspec]) => {
+      Logger.println('Pushing source code to Clever Cloud.');
       const s_push = Git.push(remote, branchRefspec, force);
-      return Bacon.combineAsArray(s_push, s_appData, commitIdToPush);
+      return Bacon.combineAsArray(s_push, s_appData, s_commitIdToPush);
     })
     .flatMapLatest(([push, appData, commitId]) => {
       Logger.println('Your source code has been pushed to Clever Cloud.');
@@ -123,6 +134,7 @@ function getAllLogs (api, push, appData, commitId, quiet) {
       });
     });
 
+  // TODO, could be done without a Bus (with merged streams)
   const s_allLogs = new Bacon.Bus();
 
   s_deploymentStart.onValue((e) => {

--- a/src/commands/deploy.js
+++ b/src/commands/deploy.js
@@ -91,7 +91,7 @@ function restart (api, params) {
 };
 
 function handleError (error) {
-  Logger.error(colors.bold.red(_.get(error, 'message', error)));
+  Logger.error(_.get(error, 'message', error));
   process.exit(1);
 }
 

--- a/src/logger.js
+++ b/src/logger.js
@@ -1,9 +1,14 @@
 'use strict';
 
 const _ = require('lodash');
+const colors = require('colors/safe');
 
 function getPrefix (severity) {
-  return `[${severity.toUpperCase()}] `;
+  const prefix = `[${severity.toUpperCase()}] `;
+  if (severity === 'error') {
+    return colors.bold.red(prefix);
+  }
+  return prefix;
 }
 
 function processApiError (error) {

--- a/src/logger.js
+++ b/src/logger.js
@@ -5,10 +5,11 @@ const colors = require('colors/safe');
 
 function getPrefix (severity) {
   const prefix = `[${severity.toUpperCase()}] `;
+  const prefixLength = prefix.length;
   if (severity === 'error') {
-    return colors.bold.red(prefix);
+    return { prefix: colors.bold.red(prefix), prefixLength };
   }
-  return prefix;
+  return { prefix, prefixLength };
 }
 
 function processApiError (error) {
@@ -19,15 +20,23 @@ function processApiError (error) {
   return [`${error.message} [${error.id}]`, ...fields].join('\n');
 };
 
+function formatLines (prefixLength, lines) {
+  const blankPrefix = _.repeat(' ', prefixLength);
+  return _(lines)
+    .split('\n')
+    .map((line, i) => (i === 0) ? line : `${blankPrefix}${line}`)
+    .join('\n');
+}
+
 const Logger = _(['debug', 'info', 'warn', 'error'])
   .map((severity) => {
     if (process.env['CLEVER_QUIET'] || !process.env['CLEVER_VERBOSE'] && (severity === 'debug' || severity === 'info')) {
       return [severity, _.noop];
     }
     const consoleFn = (severity === 'error') ? console.error : console.log;
-    const prefix = getPrefix(severity);
+    const { prefix, prefixLength } = getPrefix(severity);
     return [severity, (message) => {
-      const formattedMsg = processApiError(message);
+      const formattedMsg = formatLines(prefixLength, processApiError(message));
       return consoleFn(`${prefix}${formattedMsg}`);
     }];
   })

--- a/src/logger.js
+++ b/src/logger.js
@@ -1,33 +1,41 @@
-'use strict'
+'use strict';
 
-const _ = require('lodash')
+const _ = require('lodash');
 
-const processApiError = function (error) {
-  if (error.id == null || error.message == null) {
-    return error
-  }
-  const fields = _.map(error.fields, (msg, field) => `${field}: ${msg}`)
-  return [`${error.message} [${error.id}]`, ...fields].join('\n')
+function getPrefix (severity) {
+  return `[${severity.toUpperCase()}] `;
 }
+
+function processApiError (error) {
+  if (error.id == null || error.message == null) {
+    return error;
+  }
+  const fields = _.map(error.fields, (msg, field) => `${field}: ${msg}`);
+  return [`${error.message} [${error.id}]`, ...fields].join('\n');
+};
 
 const Logger = _(['debug', 'info', 'warn', 'error'])
   .map((severity) => {
     if (process.env['CLEVER_QUIET'] || !process.env['CLEVER_VERBOSE'] && (severity === 'debug' || severity === 'info')) {
-      return [severity, _.noop]
+      return [severity, _.noop];
     }
-    const consoleFn = (severity === 'error') ? console.error : console.log
-    return [severity, (error) => consoleFn(`[${severity.toUpperCase()}]`, processApiError(error))]
+    const consoleFn = (severity === 'error') ? console.error : console.log;
+    const prefix = getPrefix(severity);
+    return [severity, (message) => {
+      const formattedMsg = processApiError(message);
+      return consoleFn(`${prefix}${formattedMsg}`);
+    }];
   })
   .fromPairs()
-  .value()
+  .value();
 
 // No decoration for Logger.println
-Logger.println = console.log
+Logger.println = console.log;
 
 // No decoration for Logger.printErrorLine
-Logger.printErrorLine = console.error
+Logger.printErrorLine = console.error;
 
 // Only exported for testing, shouldn't be used directly
-Logger.processApiError = processApiError
+Logger.processApiError = processApiError;
 
-module.exports = Logger
+module.exports = Logger;

